src/main/java/com/tstepnik/planner/exceptions/ControllerAdvisor.java

```diff
@@ -0,0 +1,79 @@
+package com.tstepnik.planner.exceptions;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import javax.persistence.EntityNotFoundException;
+import java.time.LocalDateTime;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@ControllerAdvice
+public class ControllerAdvisor extends ResponseEntityExceptionHandler {
```

Comment:
What is this class you are extending?

Reply:
I don't know it at all, I did this ControllerAdvisor according to tutorial and it's working. This absctract class has  few methods for handle exception e.g. for exception from annotation @Valid.  I removed it and app catchs exceptions as well so now I'm not sure if i even need it.

Reply:
Oh, cool, but in the future don't add stuff that you don't understand. It was probably explained in the tutorial you've used.